.github/workflows/crypto3-testing-mac.yml

```
@@ -0,0 +1,173 @@
+          echo "platform-version=${platform_version}" >> $GITHUB_OUTPUT
+
+          if [ "${{ inputs.concurrency }}" = "0" ]; then
+            echo "Setting concurrency to number of logical cores"
+            proc_number=$(sysctl -n hw.logicalcpu)
+          else
+            echo "Setting concurrency to user-defined value"
+            proc_number=${{ inputs.concurrency }}
+          fi
+
+          echo "proc-number=${proc_number}" >> $GITHUB_OUTPUT
+
+      - name: Set up dependencies directory
+        run: |
+          mkdir -p "${{ steps.strings.outputs.dependencies-dir }}"
+
+      - name: Install boost
+        uses: MarkusJx/install-boost@v2.4.4
+        id: install-boost
+        with:
+          boost_version: ${{ inputs.boost-version }}
+          # A list of supported versions can be found here:
+          # https://github.com/MarkusJx/prebuilt-boost/blob/main/versions-manifest.json
+          platform_version: ${{ steps.strings.outputs.platform-version }}
+          boost_install_dir: ${{ steps.strings.outputs.dependencies-dir }}
+
+      - name: Configure CMake
+        run: >
+          cmake -B build
+          -DCMAKE_CXX_COMPILER=${{ matrix.cpp_compiler }}
+          -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
+          -DBUILD_TESTS=TRUE
+          -DENABLE_JUNIT_TEST_OUTPUT=TRUE
+          -S ${{ github.workspace }}
+
+        env:
+          BOOST_ROOT: "${{ steps.install-boost.outputs.BOOST_ROOT }}"
+
+      - name: Build tests
+        id: build
+        working-directory: ${{ github.workspace }}/build
+        run: |
+          # Not considering failed targets bad. We will handle them as junit test result
+          build_log_path=${{ github.workspace }}/build/build.log
+          targets_str=$(echo "${{ inputs.targets }}" | awk '{$1=$1};1' | sed '/^$/d' | tr '\n' ' ' | sed 's/ $//')
+          make -k -j ${{ steps.strings.outputs.proc-number }} ${targets_str} 2>&1 | tee ${build_log_path}
+          echo "build-log=$build_log_path" >> "$GITHUB_OUTPUT"
+
+      - name: Generate JUnit Report from build result
+        id: make-build-report
+        uses: NilFoundation/ci-cd/actions/build-log-to-junit@v1
+        with:
+          build-log: ${{ github.workspace }}/build/build.log
+
+      - name: Run tests
+        id: run_tests
+        working-directory: ${{ github.workspace }}/build
+        # Not considering failed tests bad. We will compare diff instead
+        run: |
+          artifact_dir=${{ github.workspace }}/../results_for_uploading_${{github.sha}}
+          mkdir -p $artifact_dir
+          artifact_dir=$(cd ${{ github.workspace }}/../results_for_uploading_${{github.sha}} && pwd)
+          echo "artifact-dir=$artifact_dir" >> "$GITHUB_OUTPUT"
+
+          custom_tests_dir=$artifact_dir/macos-12/${{ matrix.cpp_compiler }}/${{ matrix.build_type }}
+          mkdir -p $custom_tests_dir
+
+          for dir in "${{ inputs.test-paths }}"
```

```suggestion
          test_paths="${{ inputs.test-paths }}"
          for dir in $(echo "${test_paths}" | awk 'NF {$1=$1; print}')
```
GH actions always add leading whitespaces and empty line at the end

---

Done.